src/neptune/main.d

```diff
@@ -0,0 +1,342 @@
+    }
+
+    bool ourMajorRelease ( Version v )
+    {
+        return v.major == patch_version.major &&
+               v.minor.isNull && v.patch.isNull;
+    }
+
+    // Get version tracking branches
+    auto branches = getBranches(repo)
+                      .map!(a=>Branch(Version(a.name), a.sha))
+                      .filter!(a=>!thisVersion(a) &&
+                                  (newerVersion(a) ||
+                                  ourMajorRelease(a)))
+                      .array()
+                      .sort();
+
+    writefln("Remote branches: %s", branches);
+
+    auto current_branch = getCurrentBranch();
+    auto last_tag = patch_version;
+
+    scope(exit)
+        list.local_actions ~= LocalAction(format("git checkout %s", current_branch),
+                                          format("Checkout original branch %s",
+                                             current_branch));
+
+    foreach (branch; branches)
+    {
+        list.local_actions ~=
+            LocalAction(format("git checkout -B %s %s/%s",
+                               branch, upstream, branch),
+                        format("Checkout %s locally", branch));
+        list.local_actions ~=
+            LocalAction(format(`git merge -m "Merge tag %s into %s" %s`,
+                               last_tag, branch, patch_version),
+                        format("Merge %s into %s", last_tag, branch));
+
+        list.affected_refs ~= branch.toString();
+
+        // Only make new releases on minor branches
+        if (branch.type != branch.type.Minor)
+            continue;
+
+        auto existing_version = tags.retro
+                                  .find!(a=>a.major == branch.major &&
+                                         a.minor == branch.minor);
+
+        if (existing_version.empty)
+            continue;
+
+        auto new_version = Version(Version.Type.Patch,
+                                   branch.major,
+                                   branch.minor,
+                                   existing_version.front.patch+1);
+
+        list ~= makeRelease(new_version, branch.toString);
+
+        last_tag = new_version;
+    }
+
+    return list;
+}
+
+
+/*******************************************************************************
+
+    Params:
+        release_version = release version to make
+        target = target reference for the release
+
+    Returns:
+        an actionlist element containing all the actions/refs required to do the
+        requested release
+
+*******************************************************************************/
+
+ActionList makeRelease ( Version release_version, string target )
+{
+    assert(!release_version.major.isNull &&
+           !release_version.minor.isNull &&
+           !release_version.patch.isNull);
+
+    import std.format;
+    auto v = release_version.toString();
+
+    return ActionList([LocalAction(format("git tag -m %s %s %s", v, v, target),
```

Comment:
So no taking/making release notes and putting them as the tag message for now, right?